backends-common/postgres/src/main/java/org/apache/james/backends/postgres/utils/PoolPostgresExecutor.java

```diff
@@ -0,0 +1,107 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.backends.postgres.utils;
+
+import static org.apache.james.backends.postgres.utils.PostgresExecutor.MAX_RETRY_ATTEMPTS;
+import static org.apache.james.backends.postgres.utils.PostgresExecutor.MIN_BACKOFF;
+import static org.apache.james.backends.postgres.utils.PostgresExecutor.PGSQL_DIALECT;
+import static org.apache.james.backends.postgres.utils.PostgresExecutor.SETTINGS;
+
+import java.time.Duration;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+import javax.inject.Inject;
+
+import org.apache.james.lifecycle.api.Disposable;
+import org.jooq.DSLContext;
+import org.jooq.Record;
+import org.jooq.impl.DSL;
+import org.reactivestreams.Publisher;
+
+import io.r2dbc.pool.ConnectionPool;
+import io.r2dbc.pool.ConnectionPoolConfiguration;
+import io.r2dbc.spi.Connection;
+import io.r2dbc.spi.ConnectionFactory;
+import io.r2dbc.spi.R2dbcBadGrammarException;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.util.retry.Retry;
+
+public class PoolPostgresExecutor implements Disposable {
+
+    static int INITIAL_SIZE = 10;
+    static int MAX_SIZE = 50;
```

Comment:
Isn't that a bit much?

Reply:
Should we make this configurable in `postgres.properties`?

Reply:
Can yes, we maybe should be able to fine tune it :)

Reply:
+1 for fine tunning,